src/vanilla.ts

```diff
@@ -41,7 +45,9 @@ export default function create<TState extends State>(
   const listeners: Set<StateListener<TState>> = new Set()
 
   const setState: SetState<TState> = (partial, replace) => {
-    const nextState = typeof partial === 'function' ? partial(state) : partial
+    // TODO: Fix me once https://github.com/microsoft/TypeScript/issues/37663 is resolved
+    // https://github.com/microsoft/TypeScript/issues/37663#issuecomment-759728342
+    const nextState = partial instanceof Function ? partial(state) : partial
```

Comment:
I didn't know the 37663 issue, but hit the issue behavior in other projects. If this is only about typing, can we just add a type assertion until the issue is resolved? That's what I do for all other projects.

Reply:
I was torn between type assertion and `instanceof` as well. Seems like tomayto/tomahto kind of thing to me. Fixed in https://github.com/pmndrs/zustand/pull/376/commits/9a48fa8dce62fadf75866a4a0569db857e7ab9f0

Reply:
Thanks! It will keep the bundle JS code identical.